src/cc/lib/graph/graph.cc

```diff
@@ -99,8 +99,14 @@ void Graph::GetNodeType(std::span<const NodeId> node_ids, std::span<Type> output
         }
         else
         {
-            const auto index = internal_id->second;
-            *curr_type = m_partitions[m_partitions_indices[index]].GetNodeType(m_internal_indices[index]);
+            auto index = internal_id->second;
+            size_t partition_count = m_counts[index];
+            for (size_t partition = 0; partition < partition_count; ++partition, ++index)
+            {
+                *curr_type = m_partitions[m_partitions_indices[index]].GetNodeType(m_internal_indices[index]);
+                if (*curr_type != snark::DEFAULT_NODE_TYPE)
```

Comment:
how do you handle a case when a node type is present only in one partition as `snark::DEFAULT_NODE_TYPE`? Where do we assign `default_type` to returned elements?

Reply:
If the node only exists in one partition with the default type, this means the node does not exist and nothing should be returned. The default type is handled in python I believe.